build-scripts/config_common.cmake

```diff
@@ -248,6 +248,12 @@ else ()
     add_definitions (-DWASM_DISABLE_STACK_HW_BOUND_CHECK=0)
   endif ()
 endif ()
+if (WAMR_DISABLE_BLOCK_INSN_INTERRUPT EQUAL 1)
```

Comment:
Should we initialize `WAMR_DISABLE_BLOCK_INSN_INTERRUPT` in `CMakeLists.txt` (similarly to what we do for other variables in `product-mini/platforms/linux/CMakeLists.txt`

Reply:
I was following what's done for `WAMR_DISABLE_HW_BOUND_CHECK` since `WAMR_DISABLE_BLOCK_INSN_INTERRUPT` is available on the same platforms where HW bound checks is.

Right now, it is enabled by default (as it is for HW bound checks).
Maybe for Windows we can disable them both directly in `product-mini/platforms/windows/CMakeLists.txt` instead of manually in the CI https://github.com/bytecodealliance/wasm-micro-runtime/blob/2eed50b3034b8eec531cce4a782b8c698fea00fd/.github/workflows/compilation_on_windows.yml#L97

Reply:
Actually that's not true, since windows doesn't include `runtime_lib.cmake` as other platforms do https://github.com/bytecodealliance/wasm-micro-runtime/blob/17f3375d530c394536379fc0e0247847ad756de9/product-mini/platforms/linux/CMakeLists.txt#L121
that in turn includes `config_common.cmake` https://github.com/bytecodealliance/wasm-micro-runtime/blob/17f3375d530c394536379fc0e0247847ad756de9/build-scripts/runtime_lib.cmake#L157